examples/01_random_field/07_precipitation.py

```diff
@@ -0,0 +1,114 @@
+P_gau = copy.deepcopy(srf.field)
+
+###############################################################################
+# Now we should take care of the dry periods. Therefore we simply introduce a
+# lower threshold value.
+
+threshold = 0.4
+srf.field[srf.field <= threshold] = 0.0
+P_cut = srf.field
+
+###############################################################################
+# With the above lines of code we have created a cut off Gaussian spatial
+# random field with an exponential variogram. But precipitation fields are not
+# distributed Gaussian. Thus, we will now transform the field with a box-cox
+# transformation, which is often used to account for the skewness of
+# precipitation fields. Different values have been suggested for the
+# transformation parameter lambda, but we will stick to 1/2. We call the
+# resulting field Gaussian anamorphosis.
+
+gs.transform.boxcox(srf, lmbda=0.5, shift=-1.0)
+
+###############################################################################
+# As a last step, the amount of precipitation is set. This should of course be
+# calibrated towards observations (the same goes for the threshold).
+
+amount = 3.0
+srf.field *= amount
+P_ana = srf.field
+
+###############################################################################
+# Finally we can have a look at the fields resulting from each step. For a
+# closer look, we will examine a cross section at an arbitrary location. And
+# afterwards we will create a contour plot for visual candy.
+
+fig, axs = plt.subplots(2, 2, sharex=True, sharey=True)
+
+axs[0,0].set_title('Gaussian')
+axs[0,0].plot(t, P_gau[:, 20])
+axs[0,0].set_ylabel(r'$P$ / mm')
+
+axs[0,1].set_title('Cut Gaussian')
+axs[0,1].plot(t, P_cut[:, 20])
+
+axs[1,0].set_title('Cut Gaussian Anamorphosis')
+axs[1,0].plot(t, P_ana[:, 20])
+axs[1,0].set_xlabel(r'$t$ / d')
+axs[1,0].set_ylabel(r'$P$ / mm')
+
+axs[1,1].set_title('Different Cross Section')
+axs[1,1].plot(t, P_ana[:, 10])
+axs[1,1].set_xlabel(r'$t$ / d')
+
+plt.tight_layout()
+
+fig, axs = plt.subplots(2, 2, sharex=True, sharey=True)
+
+axs[0,0].set_title('Gaussian')
+cont = axs[0,0].contourf(t, x, P_gau.T, cmap='PuBu')
+cbar = fig.colorbar(cont, ax=axs[0,0])
+cbar.ax.set_ylabel(r'$P$ / mm')
+axs[0,0].set_ylabel(r'$x$ / km')
+
+axs[0,1].set_title('Cut Gaussian')
+cont = axs[0,1].contourf(t, x, P_cut.T, cmap='PuBu')
+cbar = fig.colorbar(cont, ax=axs[0,1])
+cbar.ax.set_ylabel(r'$P$ / mm')
+axs[0,1].set_xlabel(r'$t$ / d')
+
+axs[1,0].set_title('Cut Gaussian Anamorphosis')
+cont = axs[1,0].contourf(t, x, P_ana.T, cmap='PuBu')
+cbar = fig.colorbar(cont, ax=axs[1,0])
+cbar.ax.set_ylabel(r'$P$ / mm')
+axs[1,0].set_xlabel(r'$t$ / d')
+axs[1,0].set_ylabel(r'$x$ / km')
+
+fig.delaxes(axs[1,1])
+plt.tight_layout()
+
+###############################################################################
+# In this example we have created precipitation fields which have one spatial
+# dimension, but it is very easy to add a second spatial dimension.
```

Comment:
There already came mails asking about how to do stuff in 2D where there was only a 1D example.. maybe we should skip this sentence or provide a minimal example for a field in 2D. What do you think?

Reply:
You are probably right. All add some lines. Hmm, maybe I can use pyvista for visualization?

Reply:
Would use a simple 2D plot with a masked array to get something like a weather forecast plot for rain for a single time-step :wink:
Or an animation? :thinking: sphinx-gallery has support for matplotlib animations: https://sphinx-gallery.github.io/stable/auto_examples/plot_8_animations.html

Reply:
Yeah, it would be pretty cool, to do something like that. I just took the easiest path with the current commits, but I'm not convinced.
I'll think about a better visualisation.

Reply:
It is also nice, that this is implicitly the first example for spatio-temporal fields with a joint model! I guess we should start here for supporting spatio-temporal tools:
https://www.rdocumentation.org/packages/gstat/versions/2.0-6/topics/vgmST